### PR TITLE
[WIP]Bump statefulset maxUnavailable to beta

### PR DESCRIPTION
**Pull request by kerthcet**
#### What type of PR is this?
/kind feature
/sig apps

#### Which issue(s) this PR fixes:

Fixes https://github.com/kubernetes/enhancements/issues/961

#### Does this PR introduce a user-facing change?

```release-note
Statefulset `MaxUnavailable` is graduated to Beta.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
- [KEP]: https://github.com/kubernetes/enhancements/issues/961
```

### CODE REVIEW

File: pkg/controller/statefulset/metrics/metrics.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+			Help:           "Duration in seconds to rolling update a statefulset successfully",
+			StabilityLevel: metrics.STABLE,
+			Buckets:        metrics.ExponentialBuckets(1, 2, 10),
+		},
+	)
+)
+
+var registerMetrics sync.Once
+
+// Register registers CronjobController metrics.
```
Comment (ppbits) [inline pkg/controller/statefulset/metrics/metrics.go:42]: StatefulSetController

File: staging/src/k8s.io/api/apps/v1/types_swagger_doc_generated.go [hunk 2]
```diff
@@ -283,7 +283,7 @@ func (RollingUpdateDeployment) SwaggerDoc() map[string]string {
...
-	"maxUnavailable": "The maximum number of pods that can be unavailable during the update. Value can be an absolute number (ex: 5) or a percentage of desired pods (ex: 10%). Absolute number is calculated from percentage by rounding up. This can not be 0. Defaults to 1. This field is alpha-level and is only honored by servers that enable the MaxUnavailableStatefulSet feature. The field applies to all pods in the range 0 to Replicas-1. That means if there is any unavailable pod in the range 0 to Replicas-1, it will be counted towards MaxUnavailable.",
+	"maxUnavailable": "The maximum number of pods that can be unavailable during the update. Value can be an absolute number (ex: 5) or a percentage of desired pods (ex: 10%). Absolute number is calculated from percentage by rounding up. This can not be 0. Defaults to 1. The field applies to all pods in the range 0 to Replicas-1. That means if there is any unavailable pod in the range 0 to Replicas-1, it will be counted towards MaxUnavailable.\n\nThis field is beta-level and is only honored by servers that enables the MaxUnavailableStatefulSet feature gate(enabled by default).",
 }
```
Comment (ppbits) [inline staging/src/k8s.io/api/apps/v1/types_swagger_doc_generated.go:287]: We may have more places with "alpha-level" to be replaced with "beta-level":
https://github.com/search?q=repo%3Akubernetes%2Fkubernetes+MaxUnavailableStatefulSet+feature&type=code

Comment (kerthcet) [reply staging/src/k8s.io/api/apps/v1/types_swagger_doc_generated.go:287]: I'll update with other stuff together, but thanks a lot.

**Comment by kerthcet** [comment]
in flavor of https://github.com/kubernetes/enhancements/pull/4474
/close

**Pull request closed without merging**